### PR TITLE
Disable client side phishing component and model

### PR DESCRIPTION
**Pull request by fmarier**
Resolves brave/brave-browser#16379

## Submitter Checklist:

- [x] I confirm that no security/privacy review [is needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or that I have [requested](https://github.com/brave/security/issues/new/choose) one
- [x] There is a [ticket](https://github.com/brave/brave-browser/issues) for my issue
- [x] Used Github [auto-closing keywords](https://docs.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue) in the PR description above
- [x] Wrote a good [PR/commit description](https://google.github.io/eng-practices/review/developer/cl-descriptions.html)
- [x] Added appropriate labels (`QA/Yes` or `QA/No`; `release-notes/include` or `release-notes/exclude`; `OS/...`) to the associated issue
- [ ] Checked the PR locally: `npm run test -- brave_browser_tests`, `npm run test -- brave_unit_tests`, `npm run lint`, `npm run gn_check`, `npm run tslint`
- [ ] Ran `git rebase master` (if needed)

## Reviewer Checklist:

- [ ] A security review [is not needed](https://github.com/brave/brave-browser/wiki/Security-reviews), or a link to one is included in the PR description
- [ ] New files have MPL-2.0 license header
- [ ] Adequate test coverage exists to prevent regressions
- [ ] Major classes, functions and non-trivial code blocks are well-commented
- [ ] Changes in component dependencies are properly reflected in `gn`
- [ ] Code follows the [style guide](https://chromium.googlesource.com/chromium/src/+/HEAD/styleguide/c++/c++.md)
- [ ] Test plan is specified in PR before merging

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on
- [ ] All relevant documentation has been updated, for instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-Compatibility-Exceptions-in-Brave
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide
  - [ ] https://github.com/brave/brave-browser/wiki/P3A

## Test Plan:

1. I ran through all of the Webpage and Desktop Download tests on <https://testsafebrowsing.appspot.com/> successfully.
2. I also created a new profile and started the browser with mitmproxy, confirming that there weren't any requests:
   - for the `imefjhfbkmcmebodilednhmaccmincoa` extension
   - to `static.brave.com` (note: `static1.brave.com` is okay) or `ssl.gstatic.com`

### CODE REVIEW

File: patches/components-safe_browsing-content-browser-client_side_model_loader.cc.patch [hunk 1]
```diff
@@ -1,13 +1,12 @@
...
--    "https://ssl.gstatic.com/safebrowsing/csd/";
-+    "https://static.brave.com/safebrowsing/csd/";
- const char ModelLoader::kClientModelNamePattern[] =
-     "client_model_v5%s_variation_%d.pb";
- #if !BUILDFLAG(FULL_SAFE_BROWSING)
+@@ -164,6 +164,7 @@ void ModelLoader::StartFetch(bool only_from_cache) {
+   // Start fetching the model either from the cache or possibly from the
+   // network if the model isn't in the cache.
+ 
++  BRAVE_DISABLE_FETCH
```
Comment (goodov) [inline patches/components-safe_browsing-content-browser-client_side_model_loader.cc.patch:9]: we can block a network request here without introducing the patch. Can be done like this:
```
#include "components/safe_browsing/content/browser/client_side_model_loader.h"

#include "net/traffic_annotation/network_traffic_annotation.h"

#define NetworkTrafficAnnotationTag \
  IsCertificateError(net::OK);      \
  return;                           \
  net::NetworkTrafficAnnotationTag

#include "../../../../../../components/safe_browsing/content/browser/client_side_model_loader.cc"

#undef NetworkTrafficAnnotationTag
```

Comment (fmarier) [reply patches/components-safe_browsing-content-browser-client_side_model_loader.cc.patch]: Thanks @goodov , that's a very neat trick. Patch has been removed.

**Review by goodov — APPROVED** [review]

File: chromium_src/components/safe_browsing/content/browser/client_side_model_loader.cc [hunk 2]
```diff
@@ -0,0 +1,17 @@
+/* Copyright (c) 2021 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "components/safe_browsing/content/browser/client_side_model_loader.h"
```
Comment (bridiver) [inline chromium_src/components/safe_browsing/content/browser/client_side_model_loader.cc:6]: these includes (one below as well) aren't really what we want here. We should be including the things used in this file which are `net/traffic_annotation/network_traffic_annotation.h` and (maybe if `IsCertificateError` is actually needed) `net/base/net_errors.h`

Comment (fmarier) [reply chromium_src/components/safe_browsing/content/browser/client_side_model_loader.cc]: I pruned the headers and added a comment for why `IsCertificateError()` is needed.

**Review by bridiver — APPROVED** [review]

**Pull request merged**